compiler/p/codegen/OMRMemoryReference.cpp

```diff
@@ -1683,10 +1700,10 @@ OMR::Power::MemoryReference::checkRegisters(TR::CodeGenerator *cg)
       }
```

Comment:
these patch-over in expandInstruction() routine look very hacky to me.
depending on the opcode (assuming the precisely optimal opcode is used in Instruction Selection phase, for example, choosing between lwz/lwzx/plwz) is not a good design.
plus, you didn't do anything substantial in routine addToOffset(). It is expanded or converted already assuming 16-bit displacement, prior coming to here.

Reply:
This PR is focused solely on getting prefixed instructions to run through the backend (i.e. instruction expansion and binary encoding) assuming that they were specifically generated in instruction selection, which requires that these instructions not be incorrectly expanded if explicitly used. Improvements regarding anything to do with instruction selection (incl. changes to `addToOffset` and proper selection between the various available instructions) will be handled in a future PR.

Reply:
it is a concern that a logically unit of work is delivered this way without a higher-level single-view design, in case this kind of hacky code is forgotten in later stages of delivery.  it will be helpful to see a single higher-level design first (e.g.  what is in plan for staged delivery where we can see top-down information as to the final form of implementation).